pkg/kube/unstructured/unstructured_test.go

```diff
@@ -519,6 +519,81 @@ func TestResourceShouldConvergeToSelector(t *testing.T) {
 	}
 }
 
+func TestResourceShouldConvergeToField(t *testing.T) {
```

Comment:
Could you add another test case with the selector containing multiple arrays and different indexes?
e.g. `.spec.template.containers[0].ports[1].containerPort=8490`